certbot/plugins/standalone.py

```diff
@@ -114,39 +112,53 @@ def running(self):
 SUPPORTED_CHALLENGES = [challenges.TLSSNI01, challenges.HTTP01]
 
 
-def supported_challenges_validator(data):
-    """Supported challenges validator for the `argparse`.
+class SupportedChallengesAction(argparse.Action):
+    """Action class for parsing standalone_supported_challenges."""
 
-    It should be passed as `type` argument to `add_argument`.
+    def __call__(self, parser, namespace, values, option_string=None):
```

Comment:
Is there a reason we weren't just doing this in the first place? It seems to me that all instances of `set_by_cli` could be avoided by using actions rather than sticking logic into `type`.